### PR TITLE
Consider MGP in eth_getPrice

### LINKED ISSUE #1343: Wrong computation in eth_gasPrice()
https://github.com/rsksmart/rskj/blob/57ba47e28967a4c35c7a0908e046d67c4774194c/rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java#L87

It seems that the returned value only considers the sliding window, but it doesn't consider the MGP.

The returned value should be: 
Math.max(lastVal,bestBlockGasPrice*11 /10); // 10% added to MGP

### PR DESCRIPTION
**Pull request by Vovchyk**
Considering best block's MGP in `eth_getPrice`.

## Description
This change takes into account best block's MGP while calculating gas price in the `eth_getPrice` JSON-RPC method. This resolves #1343

## How Has This Been Tested?

Using unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java [hunk 1]
```diff
@@ -61,30 +75,39 @@ public void onBlock(Block block, List<TransactionReceipt> receipts) {
         logger.trace("End onBlock");
     }
 
-    public void onTransaction(Transaction tx) {
+    private void onTransaction(Transaction tx) {
```
Comment (donequis) [inline rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java:78]: Empty blocks or unusued block space should reduce the suggested gas price, which is not happening with this formula. Probably an idea to track in the backlog, nothing actually related to this task

File: rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java [hunk 2]
```diff
@@ -61,30 +75,39 @@ public void onBlock(Block block, List<TransactionReceipt> receipts) {
...
+            Coin bestBlockPrice = bestBlockPriceRef.get();
+            if (bestBlockPrice == null) {
+                return lastVal;
+            } else {
+                return max(lastVal, bestBlockPrice.multiply(BI_11).divide(BI_10));
+            }
         }
     }
+
+    private static Coin max(Coin a, Coin b) {
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java:110]: wouldnt be better to declare max() in Coin? would be consistent with BigInteger.max()

**Review by patogallaiovlabs — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java (hunk 2 above)
Comment (patogallaiovlabs) [reply rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java]: nice :)

**Review by patogallaiovlabs — DISMISSED** [review]
LGTM

**Comment by jjtechuy** [comment]
pipeline: run

**Review by ajlopezrsk — APPROVED** [review]

**Pull request merged**